client/src/pages/Question/Question.tsx

```diff
@@ -0,0 +1,136 @@
+import { useParams,useNavigate, useLocation } from "react-router-dom";
+import { useEffect, useReducer, useState } from "react";
+import { State,Action,Params,Location } from "./Question.types";
+import { nanoid } from "nanoid";
+
+import "./Question.css";
+import { useQuiz } from "../../contexts/quiz.context";
+import { getLeaderboard } from "../../services/quiz.services";
+
+const Question = () => {
+    const {quizID}:Params = useParams();
+
+    const {quiz} = useQuiz();
+    const currentQuiz = quiz?.find( q => q._id === quizID )
+
+    const [time,setTime] = useState<number>(30);
+    const [stopTime,setStopTime] = useState<boolean>(false);
+    const navigate = useNavigate();
+
+    const location:Location = useLocation();
+    const state = location.state;
+
+    useEffect(() => {
+       let timer = 30;
+       function timeout(){
+          setIsClicked(true);
+          setStopTime(true)
+       }
+
+       let a = setInterval(() => {
+          if(timer === 1){
+            clearInterval(a)
+            timeout();
+          }
+          setTime(state => state - 1)
+          timer = timer - 1;
+       },1000)
+
+       if(stopTime){
+          clearInterval(a)
+       }
+
+       return () => {
+          clearInterval(a)
+       }
+    },[stopTime])
+
+    
+    const [isClicked,setIsClicked] = useState<boolean|number>(false);
+
+    const initialState:State = {
+       score:0,
+       currentQuestionNo:0
+    }
+
+    const quizReducer = (state:State,action:Action) => {
+        switch (action.type) {
+            case "INCREMENT_SCORE":
+            return {...state,score:action.payload.score}
+            case "NEXT_QUESTION":
+            return {...state,currentQuestionNo:action.payload.currentQuestionNo}
+            default:
+            return state;
+        }
+    }
+
+    const [{score,currentQuestionNo},dispatch] = useReducer(quizReducer,initialState);
+
+    function handleOption(isRight:boolean,i:number){
+        if(isRight){
+            setIsClicked(true);
+            setStopTime(true);
+            dispatch({type:"INCREMENT_SCORE",payload:{score:score + 1}});
```

Comment:
You can do the calculation in the payload so that you can test it

Reply:
done